src/polling_util.hpp

```diff
@@ -108,8 +108,8 @@ typedef int timeout_t;
 
 timeout_t
 compute_timeout (bool first_pass_, long timeout_, uint64_t now_, uint64_t end_);
-
-#elif defined ZMQ_POLL_BASED_ON_SELECT
+#endif
```

Comment:
this does not look right

Reply:
I think there are checks to exclude the possibility of ZMQ_POLL_BASED_ON_POLL and ZMQ_POLL_BASED_ON_SELECT being defined at the same time, so I think it makes no difference for those cases. I changed it like this, because ZMQ_POLL_BASED_ON_POLL can be true at the same time as ZMQ_HAVE_PPOLL, because ppoll is currently based purely on pselect, so it is independent of the poller chosen for existing poll functions.

Reply:
I made sure that the `elif` behavior is still obeyed by replacing the following `if` clause with `#if (!defined ZMQ_POLL_BASED_ON_POLL && defined ZMQ_POLL_BASED_ON_SELECT) || defined ZMQ_HAVE_PPOLL`.

Note that the elif is not possible, because then the block is always skipped when `ZMQ_POLL_BASED_ON_POLL`, but the block is also necessary in case of `ZMQ_HAVE_PPOLL`.